ChronoKeeper/KeeperRecordingService.h

```diff
@@ -0,0 +1,76 @@
+#ifndef KEEPER_RECORDING_SERVICE_H
+#define KEEPER_RECORDING_SERVICE_H
+
+#include <iostream>
+#include <margo.h>
+#include <thallium.hpp>
+#include <thallium/serialization/stl/string.hpp>
+#include "KeeperIdCard.h"
+#include "LogIngestionQueue.h"
+
+namespace tl = thallium;
+
+
+namespace chronolog
+{
+
+//#INNA :  for now assume all identifiers are uint64_t , revisit later 
+typedef uint64_t StoryId;
+typedef uint64_t StorytellerId;
+
+
+class KeeperRecordingService : public tl::provider<KeeperRecordingService> 
+{
+public:
+    
+// KeeperRecordingService should be created on the heap not the stack thus the constructor is private...
+
+    static KeeperRecordingService * CreateKeeperRecordingService(tl::engine& tl_engine, uint16_t service_provider_id
+		    , LogIngestionQueue & ingestion_queue)
+    {
+          return  new KeeperRecordingService( tl_engine, service_provider_id, ingestion_queue );
+    }  
+
+    ~KeeperRecordingService() 
+    {
+        get_engine().pop_finalize_callback(this);
+    }
+
+private:
+
+    void record_event(tl::request const& request, 
+                       StorytellerId teller_id,  StoryId story_id, 
+		       uint64_t timeStamp, const std::string& record) 
+    {
+        std::cout << "recording {"<< teller_id<<":"<<story_id<<":"<< record <<"}"<< std::endl;
```

Comment:
You should use a lock for the queue because the keeper process can have multiple service threads and they may enqueue or dequeue simultaneously.

Reply:
This is a valid comment, and I can add the lock , though the IngestionQueue is just a dummy stub for now that I'm already in the process of rewriting as part of Issue#33